### PR TITLE
Mention the template name conflict issue in blueprint templates docs

### PR DESCRIPTION
**Pull request by RaHus**
re #266

### CODE REVIEW

**Comment by ThiefMaster** [comment]
The commit message is overly long (first line shouldn't be more than 51 chars) and the reference to the issue doesn't seem to be correct (considering that github doesn't link it)

File: docs/blueprints.rst [hunk 1]
```diff
@@ -176,16 +176,24 @@ the `template_folder` parameter to the :class:`Blueprint` constructor::
 
     admin = Blueprint('admin', __name__, template_folder='templates')
 
-As for static files, the path can be absolute or relative to the blueprint
-resource folder.  The template folder is added to the searchpath of
+For static files, the path can be absolute or relative to the blueprint
+resource folder.  
+
+The template folder is added to the searchpath of
```
Comment (ThiefMaster) [inline docs/blueprints.rst]: "search path"

**Comment by RaHus** [comment]
@ThiefMaster Noted, i'll fix it on a new PR. Does the content seem ok?

**Comment by ThiefMaster** [comment]
No need for a new PR. Simply amend the commit and force-push.

File: docs/blueprints.rst [hunk 2]
```diff
@@ -176,16 +176,24 @@ the `template_folder` parameter to the :class:`Blueprint` constructor::
...
-provides in the actual application.
+For static files, the path can be absolute or relative to the blueprint
+resource folder.  
+
+The template folder is added to the search path of templates but with a lower 
+priority than the actual application's template folder. That way you can 
+easily override templates that a blueprint provides in the actual application. 
+This also means that you'll have to make sure that no blueprint template shares 
+the same relative path with any other blueprint or actual application template 
+else a name conflict will occur due to the way flask collects references to
```
Comment (jeffwidman) [inline docs/blueprints.rst]: The way this is currently worded might imply:
1. the "name conflict" will throw an exception
2. You must always make sure the names are unique, when in fact you might (rarely) wish to leverage this behavior. 

Maybe instead say something like:
> This also means that if you don't want a blueprint template to be accidentally overridden, make sure that no other blueprint or actual application template has the same relative path.

File: docs/blueprints.rst [hunk 3]
```diff
@@ -176,16 +176,24 @@ the `template_folder` parameter to the :class:`Blueprint` constructor::
...
-As for static files, the path can be absolute or relative to the blueprint
-resource folder.  The template folder is added to the searchpath of
-templates but with a lower priority than the actual application's template
-folder.  That way you can easily override templates that a blueprint
-provides in the actual application.
+For static files, the path can be absolute or relative to the blueprint
+resource folder.  
+
+The template folder is added to the search path of templates but with a lower 
+priority than the actual application's template folder. That way you can
```
Comment (jeffwidman) [inline docs/blueprints.rst]: What about adding an additional sentence here explaining how the priority order is resolved when two blueprint templates conflict?

**Comment by jeffwidman** [comment]
Big fan of this PR. There's plenty comments on SO and the issue tracker from people surprised by this behavior, so it's great to further clarify it.

File: docs/blueprints.rst (hunk 3 above)
Comment (RaHus) [inline docs/blueprints.rst]: In  [flask.templating.DispatchingJinjaLoader#_iter_loaders](https://github.com/pallets/flask/blob/master/flask/templating.py#L87) looks like the first loader checked is the main app loader and then it walks the blueprint loaders by the order they were registered

Also [this answer](http://stackoverflow.com/a/8737529/1631135) suggests the same.

> First it checks if the app's global loader, which is app.jinja_loader can locate the file. Failing that, it checks all application blueprints (in order of registration, AFAIK) for blueprint.jinja_loader in an attempt to locate the file. 

But i am still wondering about how i should phrase this

File: docs/blueprints.rst (hunk 2 above)
Comment (RaHus) [inline docs/blueprints.rst]: That is much better, thank you for your input!

**Pull request merged**